src/ui/common/src/components/workflows/WorkflowSettings.tsx

```diff
@@ -174,6 +175,45 @@ const PeriodicScheduleSelector: React.FC<PeriodicScheduleSelectorProps> = ({
   );
 };
 
+type RetentionPolicyProps = {
+  retentionPolicy?: RetentionPolicy;
+  setRetentionPolicy: (p?: RetentionPolicy) => void;
+};
+
+const RetentionPolicySelector: React.FC<RetentionPolicyProps> = ({
+  retentionPolicy,
+  setRetentionPolicy,
+}) => {
+  let value = '';
+  let helperText: string = undefined;
+  if (!retentionPolicy || retentionPolicy.k_latest_runs <= 0) {
+    helperText = 'We keep all versions of this workflow.';
+  } else {
+    value = retentionPolicy.k_latest_runs.toString();
+  }
+
+  return (
+    <TextField
+      size="small"
+      label="Number of latest versions to keep."
```

Comment:
Nit: "The number of workflow versions to retain. Older versions will be garbage collected."